invenio_users_resources/services/schemas.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 TU Wien.
+#
+# Invenio-Requests is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""User and user group schemas."""
+
+from flask_babelex import lazy_gettext as _
+from marshmallow import Schema, ValidationError, fields, missing, pre_load, \
+    validates
+from marshmallow_utils.fields import Links
+
+
+class BaseSchema(Schema):
+    """Base schema for both users and user groups."""
+
+    id = fields.String()
+    revision_id = fields.Integer(
+        dump_only=True
+    )  # needs to take both account and profile into consideration
+    links = Links(dump_only=True)
+
+    @pre_load
+    def clean(self, data, **kwargs):
+        """Removes dump_only fields.
+
+        Why: We want to allow the output of a Schema dump, to be a valid input
+             to a Schema load without causing strange issues.
+        """
+        for name, field in self.fields.items():
+            if field.dump_only:
+                data.pop(name, None)
+        return data
+
+
+class UserAccessSchema(Schema):
+    """Schema for user access."""
+
+    visibility = fields.String()
+    email_visibility = fields.String()
+
+    def validate_visibility_field(self, field_name, value):
+        """Check if the value is a valid visibility setting."""
+        if value not in ["public", "restricted"]:
+            raise ValidationError(
+                _(f"{field_name} must be either 'public' or 'restricted'")
+            )
+
+    @validates("visibility")
+    def validate_visibility(self, value):
+        """Validate the visibility field."""
+        self.validate_visibility_field("visibility", value)
+
+    @validates("email_visibility")
+    def validate_visibility(self, value):
+        """Validate the email_visibility field."""
+        self.validate_visibility_field("email_visibility", value)
```

Comment:
Given that the marshmallow error will be scoped to the field, we can skip mentioning the `field_name` in the error:

```suggestion

def validate_visibility(value):
    """Check if the value is a valid visibility setting."""
     if value not in ["public", "restricted"]:
            raise ValidationError(
                _("Value must be either 'public' or 'restricted'.")
            )
    
class UserAccessSchema(Schema):
    """Schema for user access."""
            
    visibility = fields.String(validate=validate_visibility)
    email_visibility = fields.String(validate=validate_visibility)
```